packages/lib/modules/pool/actions/add-liquidity/form/TokenInputs.tsx

```diff
@@ -31,9 +31,8 @@ export function TokenInputs({ tokenSelectDisclosureOpen, customSetAmountIn }: Pr
 
   function weightFor(tokenAddress: string) {
     return (
-      pool.poolTokens.find(token =>
-        isSameAddress(token.address as Address, tokenAddress as Address)
-      )?.weight ?? undefined
```

Comment:
This doesn't work because if the tokenAddress in question is an underlying token it doesn't find the weight.